Applications/Spire/Source/Ui/RecentColors.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Spire/Ui/RecentColors.hpp"
+#include <algorithm>
+
+using namespace Spire;
+
+const RecentColors& RecentColors::get_default_colors() {
+  static auto default_colors = [] {
+    auto colors = RecentColors();
+    colors.add_color(Qt::white);
+    colors.add_color(Qt::black);
+    colors.add_color(Qt::darkGray);
+    colors.add_color(Qt::magenta);
+    colors.add_color(Qt::yellow);
+    colors.add_color(Qt::cyan);
+    colors.add_color(Qt::blue);
+    colors.add_color(Qt::green);
+    colors.add_color(Qt::red);
+    return colors;
+  }();
+  return default_colors;
+}
+
+void RecentColors::add_color(const QColor& color) {
+  auto iter = std::find_if(m_recent_colors.begin(), m_recent_colors.end(),
+    [&] (const auto& element) { return element == color; });
+  if(iter != m_recent_colors.end()) {
+    m_recent_colors.erase(iter, std::min(iter + 1, m_recent_colors.end()));
```

Comment:
Justify the use of ```std::min``` here.

Reply:
I can't. If this line is hit then that means iter + 1 will be at most equal to m_recent_colors.end().